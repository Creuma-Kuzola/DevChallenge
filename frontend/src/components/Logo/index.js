import React from 'react';

import { StyledLink } from '../Header/styled';

import * as S from './styled';

function Logo() {
    return (
        <S.Title>
            <StyledLink to="/" exact>
                Dev<span>Challenge</span>
            </StyledLink>
        </S.Title>
    );
}

export default Logo;
