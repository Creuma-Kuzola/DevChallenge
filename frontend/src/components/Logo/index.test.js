import React from 'react';
import { render, screen } from '../../utils/test-utils';

import Logo from '.';

describe('<Logo />', () => {
    it('should render as link to home', () => {
        const { container } = render(<Logo />);

        expect(
            screen.getByRole('link', { name: /dev challenge/i })
        ).toHaveAttribute('href', '/');

        expect(container.firstChild).toMatchSnapshot();
    });
});
