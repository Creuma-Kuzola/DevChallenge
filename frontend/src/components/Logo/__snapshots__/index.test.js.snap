// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render as link to home 1`] = `
<div
  class="sc-Axmtr kaMUpM"
>
  <a
    aria-current="page"
    class="sc-AxgMl dba-Dlw active"
    href="/"
  >
    Dev
    <span>
      Challenge
    </span>
  </a>
</div>
`;
